infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -408,26 +398,27 @@ public void handleHeartbeatRequest(CorfuMsg msg, ChannelHandlerContext ctx, ISer
     public synchronized CorfuRuntime getCorfuRuntime() {
 
         if (corfuRuntime == null) {
-            corfuRuntime = new CorfuRuntime();
-            if ((Boolean) opts.get("--enable-tls")) {
-                corfuRuntime.enableTls((String) opts.get("--keystore"),
-                        (String) opts.get("--keystore-password-file"),
-                        (String) opts.get("--truststore"),
-                        (String) opts.get("--truststore-password-file"));
-                if ((Boolean) opts.get("--enable-sasl-plain-text-auth")) {
-                    corfuRuntime.enableSaslPlainText(
-                            (String) opts.get("--sasl-plain-text-username-file"),
-                            (String) opts.get("--sasl-plain-text-password-file"));
-                }
-            }
+            CorfuRuntimeParameters parameters = CorfuRuntimeParameters.builder()
+                    .tlsEnabled((Boolean)opts.get("--enable-tls"))
+                    .keyStore((String) opts.get("--keystore"))
+                    .ksPasswordFile((String) opts.get("--keystore-password-file"))
+                    .trustStore((String) opts.get("--truststore"))
+                    .tsPasswordFile((String) opts.get("--truststore-password-file"))
+                    .saslPlainTextEnabled((Boolean) opts.get("--enable-sasl-plain-text-auth"))
+                    .usernameFile((String) opts.get("--sasl-plain-text-username-file"))
+                    .passwordFile((String) opts.get("--sasl-plain-text-password-file"))
+                    // Ensure we can't reconfigure other clusters
+                    .clusterId(getCurrentLayout().getClusterId())
```

Comment:
This is kind of strange, why is the cluster id in the runtime parameters. I think we also need Server options.

Reply:
It's not strange at all.
A client can assert that they only want to connect to a certain cluster.

It probably should not be in server options, except in the single server mode.

Reply:
Its also described in the Javadoc...

Reply:
nvm, I thought this was being used to set the id in ServerContext.